abipy/core/testing.py

```diff
@@ -118,12 +120,104 @@ def has_mongodb(host='localhost', port=27017, name='mongodb_test', username=None
     except:
         return False
 
+
 def straceback():
     """Returns a string with the traceback."""
     import traceback
     return traceback.format_exc()
 
 
+def input_equality_check(ref_file, input2, rtol=1e-05, atol=1e-08, equal_nan=False):
+    """
+    function to compare two inputs
+    ref_file takes the path to reference input in json: json.dump(input.as_dict(), fp, indent=2)
+    input2 takes an AbinintInput object
+    tol relative tolerance for floats
+    we check if all vars are uniquely present in both inputs and if the values are equal (integers, strings)
+    or almost equal (floats)
+    """
+
+    from abipy.abio.inputs import AbinitInput
+
+    def check_int(i, j):
+        return i != j
+
+    def check_float(x, y):
+        return not numpy.isclose(x, y, rtol=rtol, atol=atol, equal_nan=equal_nan)
+
+    def check_str(s, t):
+        return s != t
+
+    def check_var(v, w):
+        _error = False
+        if isinstance(v, int):
+            _error = check_int(v, w)
+        elif isinstance(v, float):
+            _error = check_float(v, w)
+        elif isinstance(v, (str, unicode)):
+            _error = check_str(v, w)
+        return _error
+
+    def flatten_var(o, tree_types=(list, tuple, numpy.ndarray)):
+        flat_var = []
+        if isinstance(o, tree_types):
+            for value in o:
+                for sub_value in flatten_var(value, tree_types):
+                    flat_var.append(sub_value)
+        else:
+            flat_var.append(o)
+        return flat_var
+
+    with open(ref_file) as fp:
```

Comment:
There's a problem with the pseudos more specifically with the value of pseudo.filepath stored 
in json file that is actually the absolute path used on your machine. See https://travis-ci.org/abinit/abipy/jobs/214653916

You have to preprocess the dictionary returned by json, in particular the entries 
with the pseudopotential paths: extract the `pseudo_basename` from the string and then use:

    pseudo_path = os.path.join(abidata.dirpath, "pseudos", pseudo_basename)